fix(ui): Enable row selection on FirestoreDataTable

grlduarte opened the pull request:
## Description

When using `FirestoreDataTable` row selection is available only if `canDeleteItems = true`, but that's not always desired. For example, if you want to execute another actions when clicking a given data row (*e.g.* navigate to another view).

This PR enables selection when `canDeleteItems` is true or if `onSelectedRows` is given.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [X] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [X] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [X] I read and followed the [Flutter Style Guide].
- [X] I signed the [CLA].
- [X] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

djsjr commented:
@lesnitsky Can we have an update on this? I need this functionality, and was surprised to find you cannot select items unless you allow deletion

russellwheatley commented:
Is this something worthwhile, @lesnitsky?

grlduarte commented:
> This should go to the `packages/firebase_ui_firestore/lib/src/table_builder.dart`

did a rebase with master and changed files in correct places

russellwheatley commented:
@grlduarte - could you run `melos run format` in the repo, please?

grlduarte commented:
> @grlduarte - could you run `melos run format` in the repo, please? Also, merge with `master` branch. Thanks.

updated my branch to master branch, but `melos run format` didn't do anything to the files I changed

greenc0102 commented:
Any update on this feature?

russellwheatley commented:
@grlduarte - Could you merge with latest master, please? We can get this in once it's done 👍

grlduarte commented:
> @grlduarte - Could you merge with latest master, please? We can get this in once it's done +1

done :smiley:

lesnitsky commented:
@grlduarte I know this is getting ridiculous, but could you please open the same PR in [firebase/FirebaseUI-Flutter](https://github.com/firebase/FirebaseUI-Flutter) 😅?
